src/App/BlogPost/Presentation/Controller/ListAction.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Solid\App\BlogPost\Presentation\Controller;
+
+use Solid\App\BlogPost\Application\UseCase\ListBlogPostUseCase;
+use Solid\App\BlogPost\Presentation\Renderer\Renderer;
+
+final class ListAction
+{
+    public function __construct(
+        private readonly ListBlogPostUseCase $useCase,
+        private readonly Renderer $renderer,
+    ) {
+    }
+
+    public function __invoke(string $format = 'html'): void
+    {
+        $blogPosts = $this->useCase->execute();
+
+        if ('json' === $format) {
```

Comment:
[OCP] same here, this is open for modification and can be avoided by creating a dedicated class or method to do this job without asking for the format here, e.g.:
```php
$this->renderer->renderCollection($blogPosts);
```